File: frontend/src/types/route.ts
export interface DefinedRoute {
  id: string;
  name: string;

  start_point_id: string;
  pickup_point_id: string;
  dropoff_point_id: string;

  segment_ids: string[];
  qr_sequence: string[];

  gate_required: boolean;

  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/route.py
from datetime import datetime

from pydantic import BaseModel


class DefinedRoute(BaseModel):
    id: str
    name: str

    start_point_id: str
    pickup_point_id: str
    dropoff_point_id: str

    segment_ids: list[str]
    qr_sequence: list[str]

    gate_required: bool

    created_at: datetime
    updated_at: datetime | None = None

